src/main/java/htsjdk/variant/variantcontext/VariantContext.java

```diff
@@ -1233,7 +1233,21 @@ public void validateAlternateAlleles() {
             throw new TribbleException.InternalCodecException(String.format("one or more of the ALT allele(s) for the record at position %s:%d are not observed at all in the sample genotypes", getContig(), getStart()));
     }
 
+    private void validateNumberOfItems(final String attributeKey, final int observedSize ) {
+        if ( hasAttribute(attributeKey) && observedSize > 0) {
+            Object object = getAttribute(attributeKey);
+            int reportedSize = (object instanceof List ) ? ((List) object).size() : 1;
```

Comment:
I did some work on getters for attribute lists in the `VariantContext` (see #712) and I think that this implementation could throw some errors if the AC or AF are set as an array (e.g., int[]). You can use here the method `getAttributeAsList(attributeKey)` to take into account this implementation. If the attribute is a single value, the returned value is a list with size = 1, so this also will simplify the code here.